Final edit (#927)
cosmos/sdk-tutorials: academy/3-running-a-chain/node-api-and-cli.md

# Running a Node, API, and CLI

Run a blockchain and learn how to interact with it.

> There are different ways to run a node of a Cosmos blockchain. In this section you will learn how to do so using [`simapp`](https://github.com/cosmos/cosmos-sdk/tree/master/simapp).

Reviewer: ```suggestion
There are different ways to run a node of a Cosmos blockchain. In this section you will learn how to do so using [`simapp`](https://github.com/cosmos/cosmos-sdk/tree/main/simapp).
```

<HighlightBox type="tip">

To take a step back and revisit how accounts work, go to the [section on _Accounts_ in the _Main Concepts_ chapter](../2-main-concepts/accounts.md).

</HighlightBox>

You can also inspect your keys. These are held in the backend keyring which by default is that of the operating system:

```sh
$ ./simd keys list
[]
```

First add a new key:

```sh

> $ ./simd keys add academy

Reviewer 2: Again, I would suggest moving the output from the command to a different code snippet for copy-pasting purposes.

or with:

```sh
$ ./simd keys show academy
```

## Make yourself a proper validator

> As you know by now, a Cosmos SDK blockchain relies on identified validators to produce blocks. Initially there is no validator to generate blocks.

Reviewer 3: Here, I think the catch-22 comment helps understand why we bother with genesis account and transaction.
At least, for those who come from a PoW world, it would not register, and they would believe that we add these genesis elements just because this is an exercise and we want to show how those things are done if you wanted to do them. When in fact you **have** to do them, which is what the catch-22 comment drills down.

Reviewer 4: @8bitpal please check this comment. PR is approved

Reviewer 2: I think that makes sense, but would just rephrase it differently (and would exclude the word "identified"):

```suggestion
As you know by now, a Cosmos SDK blockchain relies on validators to produce blocks. Initially there is no validator to generate blocks. In order to get your chain up and running, you'll need to set up at least one validator in your genesis file, and allocate some tokens so that these can be staked and the chain can be started. This is where your tokens are initially minted, because you define how many tokens are going to be part of the genesis file.
```

> grep -A 10 balances ~/.simapp/config/genesis.json

Reviewer 2: Adding dollar sign to be consistent.

```suggestion
$ grep -A 10 balances ~/.simapp/config/genesis.json
```
